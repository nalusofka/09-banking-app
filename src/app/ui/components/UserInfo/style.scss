@use '../../styles/variables' as var;
@use '../../styles/mixins' as mix;

.user-info {
  display: grid;
  grid-template-columns: 1fr auto;
  text-decoration: inherit;
  transition: transform 0.3s ease;
  display: flex;
  align-items: center;

  &_content {
    margin-left: 20px;
    @include mix.flex;
  }
}
