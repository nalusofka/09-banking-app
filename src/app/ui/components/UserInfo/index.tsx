import React from 'react';
import Body from '../Body';
import Title from '../Title';
import Avatar from 'react-avatar';
import './style.scss';

interface UserInfoProps {
  name?: string;
  lastname?: string;
  username?: string;
}

const UserInfo: React.FC<UserInfoProps> = ({ name, lastname, username}) => {
  return (
    <div className='user-info'>
      <Avatar name={`${name} ${lastname}`} round='100px' color='#012b73' size='60' />
      <div className='user-info_content'>
        <Title as="h3" color='primary'>{name} {lastname}</Title>
        <Body color='primary'>{username}</Body>
      </div>
    </div>
  );
};

export default UserInfo;
