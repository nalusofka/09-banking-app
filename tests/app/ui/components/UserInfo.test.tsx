import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import UserInfo from '../../../../src/app/ui/components/UserInfo';

describe('Componente UserInfo', () => {
  it('renderiza correctamente con nombre, apellido y nombre de usuario', () => {
    const { getByText, asFragment } = render(
      <UserInfo name="John" lastname="Doe" username="johndoe" />
    );

    expect(getByText('John Doe')).toBeInTheDocument();
    expect(getByText('johndoe')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('aplica estilos predeterminados correctamente', () => {
    const { container, asFragment } = render(
      <UserInfo name="Jane" lastname="Smith" username="janesmith" />
    );

    const userInfo = container.querySelector('.user-info');
    expect(userInfo).toBeInTheDocument();
    expect(userInfo).toHaveClass('user-info');
    expect(asFragment()).toMatchSnapshot();
  });

  it('maneja correctamente la ausencia de propiedades opcionales', () => {
    const { container, asFragment } = render(<UserInfo />);

    const userInfo = container.querySelector('.user-info');
    expect(userInfo).toBeInTheDocument();
    expect(container).not.toHaveTextContent(/John|Doe|johndoe/i);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza correctamente solo con el nombre de usuario', () => {
    const { getByText, asFragment } = render(<UserInfo username="usernameOnly" />);

    expect(getByText('usernameOnly')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
