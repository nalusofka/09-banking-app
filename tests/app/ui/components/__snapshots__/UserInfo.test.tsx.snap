// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente UserInfo > aplica estilos predeterminados correctamente 1`] = `
<DocumentFragment>
  <div
    class="user-info"
  >
    <div
      class=" sb-avatar"
      style="display: inline-block; vertical-align: middle; width: 60px; height: 60px; border-radius: 100px; font-family: Helvetica, Arial, sans-serif;"
    >
      <div
        class=" sb-avatar__text"
        style="width: 60px; height: 60px; line-height: initial; text-align: center; color: rgb(255, 255, 255); background: rgb(1, 43, 115); border-radius: 100px;"
        title="Jane Smith"
      >
        <div
          style="display: table; table-layout: fixed; width: 100%; height: 100%;"
        >
          <span
            style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="user-info_content"
    >
      <h3
        class="title title-primary"
        data-testid="title-component"
      >
        Jane Smith
      </h3>
      <p
        class="body' body-primary"
        data-testid="body-component"
      >
        janesmith
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Componente UserInfo > maneja correctamente la ausencia de propiedades opcionales 1`] = `
<DocumentFragment>
  <div
    class="user-info"
  >
    <div
      class=" sb-avatar"
      style="display: inline-block; vertical-align: middle; width: 60px; height: 60px; border-radius: 100px; font-family: Helvetica, Arial, sans-serif;"
    >
      <div
        class=" sb-avatar__text"
        style="width: 60px; height: 60px; line-height: initial; text-align: center; color: rgb(255, 255, 255); background: rgb(1, 43, 115); border-radius: 100px;"
        title="undefined undefined"
      >
        <div
          style="display: table; table-layout: fixed; width: 100%; height: 100%;"
        >
          <span
            style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="user-info_content"
    >
      <h3
        class="title title-primary"
        data-testid="title-component"
      >
         
      </h3>
      <p
        class="body' body-primary"
        data-testid="body-component"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Componente UserInfo > renderiza correctamente con nombre, apellido y nombre de usuario 1`] = `
<DocumentFragment>
  <div
    class="user-info"
  >
    <div
      class=" sb-avatar"
      style="display: inline-block; vertical-align: middle; width: 60px; height: 60px; border-radius: 100px; font-family: Helvetica, Arial, sans-serif;"
    >
      <div
        class=" sb-avatar__text"
        style="width: 60px; height: 60px; line-height: initial; text-align: center; color: rgb(255, 255, 255); background: rgb(1, 43, 115); border-radius: 100px;"
        title="John Doe"
      >
        <div
          style="display: table; table-layout: fixed; width: 100%; height: 100%;"
        >
          <span
            style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="user-info_content"
    >
      <h3
        class="title title-primary"
        data-testid="title-component"
      >
        John Doe
      </h3>
      <p
        class="body' body-primary"
        data-testid="body-component"
      >
        johndoe
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Componente UserInfo > renderiza correctamente solo con el nombre de usuario 1`] = `
<DocumentFragment>
  <div
    class="user-info"
  >
    <div
      class=" sb-avatar"
      style="display: inline-block; vertical-align: middle; width: 60px; height: 60px; border-radius: 100px; font-family: Helvetica, Arial, sans-serif;"
    >
      <div
        class=" sb-avatar__text"
        style="width: 60px; height: 60px; line-height: initial; text-align: center; color: rgb(255, 255, 255); background: rgb(1, 43, 115); border-radius: 100px;"
        title="undefined undefined"
      >
        <div
          style="display: table; table-layout: fixed; width: 100%; height: 100%;"
        >
          <span
            style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="user-info_content"
    >
      <h3
        class="title title-primary"
        data-testid="title-component"
      >
         
      </h3>
      <p
        class="body' body-primary"
        data-testid="body-component"
      >
        usernameOnly
      </p>
    </div>
  </div>
</DocumentFragment>
`;
